pkg/metadata/annotation_deletion_propagation_policy.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metadata
+
+import (
+	"kpt.dev/configsync/pkg/api/configsync"
+	"kpt.dev/configsync/pkg/core"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DeletionPropagationPolicy is the type used to identify value enums to use
+// with the deletion-propagation-policy annotation.
+type DeletionPropagationPolicy string
+
+// String returns the string value of the DeletionPropagationPolicy.
+// Implements the Stringer interface.
+func (p DeletionPropagationPolicy) String() string {
+	return string(p)
+}
+
+const (
+	// DeletionPropagationPolicyAnnotationKey is the annotation key set on
+	// RootSync/RepoSync objects to indicate what do do with the managed
+	// resources when the RootSync/RepoSync object is deleted.
+	DeletionPropagationPolicyAnnotationKey = configsync.ConfigSyncPrefix + "deletion-propagation-policy"
+	// DeletionPropagationPolicyForeground indicates that the managed resources
+	// should all be deleted/pruned before the RootSync/RepoSync object is deleted.
+	// This will block deletion of the RootSync/RepoSync using a finalizer.
+	DeletionPropagationPolicyForeground DeletionPropagationPolicy = "Foreground"
+	// DeletionPropagationPolicyOrphan indicates that the managed resources
+	// should all be orphanned (not deleted) when the RootSync/RepoSync object
+	// is deleted.
+	// This will NOT block deletion of the RootSync/RepoSync AND will not
+	// remove or modify any config sync managed annotations.
```

Comment:
nit: this comment can be updated as this now removes CS metadata

Reply:
Done